torch/testing/_internal/distributed/multi_threaded_pg.py

```diff
@@ -32,6 +37,21 @@ def ret_work(ret):
     return _create_work_from_future(fut)
 
 
+class AllReduce:
+    def __init__(self, op):
+        assert op == ReduceOp.SUM
+        self.op = op
+
+    def work(self, data):
+        # data: List[List[Tensor]]
+        res = torch.empty_like(data[0][0]).copy_(data[0][0])
```

Comment:
why we need to create a new tensor and copy it over? couldn't we just reuse the tensor from the first rank?